services/scribe/backfill/contract.go

```diff
@@ -103,25 +104,27 @@ func (c *ContractBackfiller) Backfill(parentCtx context.Context, givenStart uint
 		return fmt.Errorf("could not get last indexed: %w", err)
 	}
 	if lastBlockIndexed > startHeight {
-		LogEvent(WarnLevel, "Using last indexed block (lastIndexBlock > startHeight)", LogData{"cid": c.chainConfig.ChainID, "sh": startHeight, "eh": endHeight})
 		startHeight = lastBlockIndexed + 1
 	}
-	LogEvent(InfoLevel, "Beginning to backfill contract ", LogData{"cid": c.chainConfig.ChainID, "sh": startHeight, "eh": endHeight})
 
 	// logsChain and doneChan are used to pass logs from rangeFilter onto the next stage of the backfill process.
-	logsChan, doneChan := c.getLogs(groupCtx, startHeight, endHeight)
+	logsChan, errChan := c.getLogs(groupCtx, startHeight, endHeight)
 
 	// Reads from the local logsChan and stores the logs and associated receipts / txs.
 	g.Go(func() error {
 		concurrentCalls := 0
 		gS, storeCtx := errgroup.WithContext(ctx)
+		// could change this to for - range
 		for {
 			select {
 			case <-groupCtx.Done():
 				LogEvent(ErrorLevel, "Context canceled while storing and retrieving logs", LogData{"cid": c.chainConfig.ChainID, "ca": c.contractConfig.Address})
 
 				return fmt.Errorf("context canceled while storing and retrieving logs: %w", groupCtx.Err())
-			case log := <-logsChan:
+			case log, ok := <-logsChan:
+				if !ok {
+					return nil
+				}
```

Comment:
No more doneChan, instead there is a listener for the channel being closed and empty. This goroutine is exited when this condition is met.

There is another channel but specifically for catching errors for better logging. The previous channel structure was susceptible to race conditions where a value from the doneChan could be received before all logs had been removed from the logsChan.